tests/v2/validation/charts/backup_restore/backup_restore_test.go

```diff
@@ -0,0 +1,118 @@
+//go:build (validation || infra.any || cluster.any || extended) && !sanity && !stress
+
+package backup_restore
+
+import (
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	bv1 "github.com/rancher/backup-restore-operator/pkg/apis/resources.cattle.io/v1"
+	"github.com/rancher/shepherd/clients/rancher"
+	"github.com/sirupsen/logrus"
+
+	"github.com/rancher/rancher/tests/v2/actions/charts"
+	"github.com/rancher/rancher/tests/v2/actions/projects"
+	"github.com/rancher/rancher/tests/v2/actions/provisioning"
+	shepCharts "github.com/rancher/shepherd/extensions/charts"
+	"github.com/rancher/shepherd/pkg/config"
+	"github.com/rancher/shepherd/pkg/session"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type BackupTestSuite struct {
+	suite.Suite
+	client   *rancher.Client
+	session  *session.Session
+	S3Client *s3.Client
+}
+
+func (b *BackupTestSuite) TearDownSuite() {
+	b.session.Cleanup()
+}
+
+func (b *BackupTestSuite) SetupSuite() {
+	b.session = session.NewSession()
+
+	client, err := rancher.NewClient("", b.session)
+	require.NoError(b.T(), err)
+
+	b.client = client
+
+	subSession := b.session.NewSession()
+	defer subSession.Cleanup()
+}
+
+func (b *BackupTestSuite) TestS3InPlaceRestore() {
+	subSession := b.session.NewSession()
+	defer subSession.Cleanup()
+
+	project, err := projects.GetProjectByName(b.client, cluster, "System")
+	require.NoError(b.T(), err)
+
+	config.LoadConfig(charts.BackupRestoreConfigurationFileKey, backupRestoreConfig)
+
+	logrus.Info("Checking if the backup chart is already installed...")
+	initialBackupChart, err := shepCharts.GetChartStatus(b.client, project.ClusterID, "cattle-resources-system", "rancher-backup")
+	require.NoError(b.T(), err)
+
+	if !initialBackupChart.IsAlreadyInstalled {
+		installBroChart(b.client)
+	}
+
+	b.client, err = b.client.ReLogin()
+	require.NoError(b.T(), err)
+
+	logrus.Info("Creating two users, projects, and role templates...")
+	userList, projList, roleList, err := createRancherResources(b.client, project.ClusterID, "cluster")
+	require.NoError(b.T(), err)
+
+	logrus.Info("Provisioning a downstream RKE1 cluster...")
+	rke1ClusterObj, _, rke1ClusterConfig, err := createDownstreamCluster(b.client, "RKE1")
+	require.NoError(b.T(), err)
+
+	logrus.Info("Provisioning a downstream RKE2 cluster...")
+	_, rke2SteveObj, rke2ClusterConfig, err := createDownstreamCluster(b.client, "RKE2")
```

Comment:
I think you should add a verifyCluster to each of these before creating the backup of the local cluster